### PR TITLE
Create an integration tests for delete category endpoint

### PR DESCRIPTION
**Pull request by Dharakkkk**
#### Checklist template for LEADERS

&#x2611; | &#x2610;

&#x2611; New code is covered with tests and coverage requirements are met.
&#x2611; Build is not failing.
&#x2611; All tests are passing.
&#x2611; All conflicts are resolved and the branch is rebased on top of the up-to-date parent branch.
&#x2611; All issue requirements are implemented.
&#x2611; At least two reviewers are assigned.
&#x2611; The Pull request is linked to the related issue.
&#x2611; Correct branch name. Convention [https://github.com/bycza-zagroda/track-expenses-app-documentation/blob/develop/githubwork/GITHUB_WORK.md]
&#x2611; Correct unit test name - suffix 'Test' in the end.
&#x2611; Correct integration test name - suffix 'IT' in the end.
&#x2611; Positive and negative test paths have been implemented.

### CODE REVIEW

**Review by dawciobiel — APPROVED** [review]
LGTM

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/repository/FinancialTransactionCategoryRepository.java [hunk 1]
```diff
@@ -17,8 +17,8 @@ public interface FinancialTransactionCategoryRepository extends JpaRepository<Fi
     @Query(nativeQuery = false, value = "select c from FinancialTransactionCategory c where c.user.id = :userId")
     Optional<List <FinancialTransactionCategory>> findAllByUserId(Long userId);
 
-    @Query(nativeQuery = false, value = "select c from FinancialTransactionCategory c where c.id = :id and c.user.id = :userId")
-    boolean existsByIdAndUserId(Long id, Long userId);
+    @Query("SELECT CASE WHEN COUNT(c) > 0 THEN true ELSE false END FROM FinancialTransactionCategory c WHERE c.id = :id and c.user.id = :userId")
```
Comment (poneciak57) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/repository/FinancialTransactionCategoryRepository.java:20]: CASE seems redundant in here, try to replace it with something like this:
```sql
SELECT COUNT(c) > 0 as result
FROM FinancialTransactionCategory c
WHERE c.id = :id AND c.user.id = :userId;
```

**Review by poneciak57 — CHANGES_REQUESTED** [review]
changes required

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/DeleteFinancialCategoryIT.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        return userRepository.save(user);
+    }
+
+    private FinancialTransactionCategory createFinancialTransactionCategory(User user) {
+        final FinancialTransactionCategory category = FinancialTransactionCategory.builder()
+                .name("Test_Name")
+                .user(user)
+                .type(FinancialTransactionType.INCOME)
+                .build();
+        return categoryRepository.save(category);
```
Comment (dawciobiel) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/DeleteFinancialCategoryIT.java:98]: One empty line can be added above the `return` statement.

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/DeleteFinancialCategoryIT.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    }
+
+    private User createTestUser() {
+        final User user = User.builder()
+                .userName("Test_Name")
+                .email("Email@mail.pl")
+                .password("Password1@")
+                .userStatus(UserStatus.VERIFIED)
+                .build();
+        return userRepository.save(user);
```
Comment (dawciobiel) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/DeleteFinancialCategoryIT.java:89]: One empty line can be added above the `return` statement.

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/DeleteFinancialCategoryIT.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    private FinancialTransactionCategoryRepository categoryRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private UserService userService;
+
+    private static final String AUTHORIZATION = "Authorization";
+    private static final String BEARER = "Bearer ";
```
Comment (dawciobiel) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/DeleteFinancialCategoryIT.java:32]:
```suggestion
    private static final String AUTHORIZATION = "Authorization";

    private static final String BEARER = "Bearer ";
```

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/DeleteFinancialCategoryIT.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+    private final Long nonExistentCategoryId = 999L;
+    private final String deleteCategoryUrl = "/api/categories/{id}";
```
Comment (dawciobiel) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/DeleteFinancialCategoryIT.java:35]:
```suggestion
    private final Long nonExistentCategoryId = 999L;

    private final String deleteCategoryUrl = "/api/categories/{id}";
```

**Review by dawciobiel — CHANGES_REQUESTED** [review]
Fix the code formatting and I'll accept it.

**Review by dawciobiel — APPROVED** [review]
LGTM

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/repository/FinancialTransactionCategoryRepository.java [hunk 6]
```diff
@@ -17,8 +17,7 @@ public interface FinancialTransactionCategoryRepository extends JpaRepository<Fi
     @Query(nativeQuery = false, value = "select c from FinancialTransactionCategory c where c.user.id = :userId")
     Optional<List <FinancialTransactionCategory>> findAllByUserId(Long userId);
 
-    @Query(nativeQuery = false, value = "select c from FinancialTransactionCategory c where c.id = :id and c.user.id = :userId")
+    @Query(value = "SELECT COUNT(c) > 0 as result FROM FinancialTransactionCategory c WHERE c.id = :id AND c.user.id = :userId")
```
Comment (poneciak57) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/repository/FinancialTransactionCategoryRepository.java:20]:
```suggestion
    @Query(nativeQuery = false, value = "SELECT COUNT(c) > 0 as result FROM FinancialTransactionCategory c WHERE c.id = :id AND c.user.id = :userId")
```

**Review by poneciak57 — CHANGES_REQUESTED** [review]

**Review by poneciak57 — APPROVED** [review]
Looks good to me

**Review by dawciobiel — APPROVED** [review]
LGTM

**Pull request merged**